### DISCUSSION TITLE
worcester heatslave ii ht4i monitoring

### DISCUSSION THREAD

**Discussion by jtricerolph** [emsesp/EMS-ESP32#1483] [Q&A]
Was hoping someone might have some sugestions.
Have a worcester heatslave ii with ht4i board and connected up fine and receiving data in HA.
Seems like not much relivent writable entities and allot of irelivent readable values or entities that don't change.
I want to monitor on time in order to calculate oil use and energy for hot water and heating.
Watching the last few days it's seems like data I will need to monitor that update at expected times is what I presume to be the oil pump (boiler_burngas) and the 3 port valve (boiler_ww3wayvalve) switching when heating is running.
It's a combi style so hot water is direct and presumeably the 3 port valve defaulting to hot water position leaves me thinking:
Boiler_burngas indicating the boiler is fired up.
Is 3 port is still 'on' its providing hot water and if on heating it running.

Does this login sound reasonable to those here with more experiance?

My understanding is oil oilers don't really modulate so once I find out the jet flow rate can calculate consumption and energy accordingly and divide that between hot water and heating using 3port valve state.

I had hoped there might be more 'usefull' entities but either because it's oil and so less common vs gas or just lest data to pull 🤔 

I was quite surprised also that can't seem to trigger the heating from ems, not end of the world as will fall back to origional plan of simple smart relay on the heating trigger circuit.
Rads are having zigbee trvs so HA will be calling for the heat.

**Comment by MichaelDvP** [comment]
For energy calculation we've added special entities, see #1148 and #1268

Modulating a oil boiler is is common for large systems (>35-50kW), for smaller ones it is too expensive, because it needs modulation of oil and fan and lambda control of both. Also oil modulation below 5 kW is not possible.
Small burners are single or double stage. You can simply calculate the energy consumpting by reading the UBA worktime for heat (heat stage 2) and dhw and nominal power of the system. For oil consuption use 9.8 kWh per liter (e.g. 15kW boiler burns 1,53 liters per hour).

To trigger heating you need to understand the boiler system, how it works and what you can set. In diagnostic mode you can set burner, pump etc, but this is not usefull for normal operation. There are a lot of threads about contolling the boiler, the special entiy `forceheatingoff` and how to control with and without thermostat.

**Reply by jtricerolph** [reply]
Thanks for the comments, will look into those threads.
I still need to direct full spec of the boiler, havnt seen the stage 2 entity change as of yet so far so guessing maybe only single burner.

I had hoped to bind the trvs heat call to a zigbee relay for redundancy on ha being down but unfortunatly seems my trvs are not compatible for binding so looks like ha route it is.

I need to have a proper look through the entities and see what's actually active as many seem idle.

**Reply by jtricerolph** [reply]
I have seen the energy kw entities however these don't seem to be doing much. Havnt been through all docs yet.
Nominal is at 19kw which I guess is default or pulled from diag port.

**Comment by jtricerolph** [comment]
@MichaelDvP I can't find how the energy entity is calculated.
I have a feeling its using fields that are not updating with my boiler.

Heating active ans dhw active dont change when running either of the corasponding services, are these the fields used to calculate energy?

**Reply by MichaelDvP** [reply]
Sure, it's possible that this boiler does not publish this info or user other telegrams. Send the support info: http://ems-esp.local/api/system/info, http://ems-esp.local/api/system/allvalues and a full log of some minutes (with the burner switing off or on in it).

**Reply by jtricerolph** [reply]
Thanks, will sort that this evening.
I did see one one of the sample images a display of heating system presure which was nice, is that common or would that have been via an external sensor?

**Reply by MichaelDvP** [reply]
Some boilers have a pressure sensor, most not. If `syspress` is missing, you don't have sensor.

**Reply by jtricerolph** [reply]
Fair enough.
I get invalid request at api/system/allvalues

**Reply by jtricerolph** [reply]
And there isn't a download allvalues button on help page

**Reply by jtricerolph** [reply]
[log (2).txt](https://github.com/emsesp/EMS-ESP32/files/13620826/log.2.txt)
[log (1).txt](https://github.com/emsesp/EMS-ESP32/files/13620827/log.1.txt)
[log.txt](https://github.com/emsesp/EMS-ESP32/files/13620828/log.txt)
[log hwater.txt](https://github.com/emsesp/EMS-ESP32/files/13620829/log.hwater.txt)

**Comment by jtricerolph** [comment]
```
{
  "System Info": {
    "version": "3.6.2-dev.2",
    "platform": "ESP32-S3",
    "uptime": "009+07:55:55.256",
    "uptime (seconds)": 806155,
    "free mem": 188,
    "max alloc": 175,
    "free app": 6096,
    "reset reason": "Power on reset / Power on reset"
  },
  "Network Info": {
    "network": "WiFi",
    "hostname": "ems-esp",
    "RSSI": -59,
    "IPv4 address": "10.88.0.126/255.255.252.0",
    "IPv4 gateway": "10.88.0.1",
    "IPv4 nameserver": "10.88.0.1",
    "static ip config": false,
    "enable IPv6": false,
    "low bandwidth": true,
    "disable sleep": true,
    "enable MDNS": true,
    "enable CORS": false,
    "AP provision mode": "disconnected",
    "AP security": "wpa2",
    "AP ssid": "ems-esp"
  },
  "NTP Info": {
    "NTP status": "connected",
    "enabled": true,
    "server": "time.google.com",
    "tz label": "Europe/London"
  },
  "OTA Info": {
    "enabled": false,
    "port": 8266
  },
  "MQTT Info": {
    "MQTT status": "connected",
    "MQTT publishes": 252917,
    "MQTT queued": 0,
    "MQTT publish fails": 0,
    "MQTT connects": 5,
    "enabled": true,
    "client id": "ems-esp",
    "keep alive": 60,
    "clean session": false,
    "entity format": 1,
    "base": "ems-esp",
    "discovery prefix": "homeassistant",
    "discovery type": 0,
    "nested format": 1,
    "ha enabled": true,
    "mqtt qos": 0,
    "mqtt retain": false,
    "publish time heartbeat": 60,
    "publish time boiler": 10,
    "publish time thermostat": 10,
    "publish time solar": 10,
    "publish time mixer": 10,
    "publish time other": 10,
    "publish time sensor": 10,
    "publish single": false,
    "publish2command": false,
    "send response": false
  },
  "Syslog Info": {
    "enabled": false
  },
  "Sensor Info": {
    "temperature sensors": 0,
    "temperature sensor reads": 0,
    "temperature sensor fails": 0,
    "analog sensors": 0,
    "analog sensor reads": 0,
    "analog sensor fails": 0
  },
  "API Info": {
    "API calls": 7,
    "API fails": 0
  },
  "Bus Info": {
    "bus status": "connected",
    "bus protocol": "HT3",
    "bus telegrams received (rx)": 555907,
    "bus reads (tx)": 80489,
    "bus writes (tx)": 1,
    "bus incomplete telegrams": 720,
    "bus reads failed": 89,
    "bus writes failed": 0,
    "bus rx line quality": 100,
    "bus tx line quality": 100
  },
  "Settings": {
    "board profile": "S32S3",
    "locale": "en",
    "tx mode": 1,
    "ems bus id": 11,
    "shower timer": false,
    "shower alert": false,
    "hide led": false,
    "notoken api": false,
    "readonly mode": false,
    "fahrenheit": false,
    "dallas parasite": false,
    "bool format": 1,
    "bool dashboard": 1,
    "enum format": 1,
    "analog enabled": true,
    "telnet enabled": true,
    "max web log buffer": 50,
    "web log buffer": 50
  },
  "Devices": [
    {
      "type": "boiler",
      "name": "GBx72/Trendline/Cerapur/Greenstar Si",
      "device id": "0x08",
      "product id": 123,
      "version": "06.02",
      "entities": 76,
      "handlers received": "0x10 0x11 0x15 0x1C 0x18 0x19 0x34 0x2A 0x04",
      "handlers fetched": "0x14 0x16 0x33 0x26",
      "handlers pending": "0xBF 0xC2 0x1A 0x35 0xD1 0xE3 0xE4 0xE5 0xE6 0xE9 0xEA",
      "handlers ignored": "0x28 0x05 0x25 0x27 0xF8"
    },
    {
      "type": "thermostat",
      "name": "Comfort RF",
      "device id": "0x18",
      "product id": 215,
      "version": "01.00",
      "entities": 7,
      "handlers received": "0x06",
      "handlers fetched": "0x02A5",
      "handlers pending": "0xA3 0xA2 0x12 0x13 0x02A6 0x02A7 0x02A8",
      "handlers ignored": "0x1A 0x35"
    },
    {
      "type": "controller",
      "name": "Controller",
      "device id": "0x09",
      "product id": 152,
      "version": "02.09",
      "entities": 0,
      "handlers ignored": "0x16 0x33"
    }
  ]
}
```

**Reply by proddy** [reply]
upgrade to the latest dev @jtricerolph - you're on a very old version

**Reply by jtricerolph** [reply]
Oops, my bad, only just received new gateway and stupidly didn't think about updating 😅

**Reply by jtricerolph** [reply]
```
{
  "Boiler Worcester GBx72/Trendline/Cerapur/Greenstar Si (DeviceID:0x08, ProductID:123, Version:06.02)": {
    "force heating off": "off",
    "heating active": "off",
    "tapwater active": "off",
    "selected flow temperature": 70,
    "heating pump modulation": 0,
    "current flow temperature": 50.3,
    "return temperature": 23.0,
    "actual boiler temperature": 52.5,
    "oil": "off",
    "gas stage 2": "off",
    "flame current": 0.0,
    "fan": "off",
    "ignition": "off",
    "oil preheating": "off",
    "burner min power": 0,
    "burner max power": 100,
    "burner min period": 1,
    "hysteresis on temperature": -6,
    "hysteresis off temperature": 4,
    "heating activated": "on",
    "heating temperature": 70,
    "heating pump": "off",
    "boiler pump max power": 100,
    "boiler pump min power": 10,
    "boiler pump mode": "proportional",
    "pump delay": 3,
    "set flow temperature": 0,
    "burner set power": 0,
    "burner selected max power": 0,
    "burner current power": 0,
    "burner starts": 24360,
    "total burner operating time": "0 days 0 hours 0 minutes",
    "burner stage 2 operating time": "0 days 0 hours 0 minutes",
    "total heat operating time": "0 days 0 hours 0 minutes",
    "burner starts heating": 11150,
    "total UBA operating time": "1408 days 20 hours 46 minutes",
    "service code number": 203,
    "maintenance message": "H00",
    "maintenance scheduled": "off",
    "time to next maintenance": 6000,
    "next maintenance date": "01.01.2009",
    "dhw turn on/off": "on",
    "dhw set temperature": 0,
    "dhw selected temperature": 59,
    "dhw solar boiler temperature": 23.1,
    "dhw type": "flow",
    "dhw comfort": "eco",
    "dhw flow temperature offset": 40,
    "dhw max power": 100,
    "dhw circulation pump available": "off",
    "dhw charging type": "3-way valve",
    "dhw hysteresis on temperature": -5,
    "dhw hysteresis off temperature": 0,
    "dhw disinfection temperature": 70,
    "dhw circulation pump mode": "2x3min",
    "dhw circulation active": "off",
    "dhw current intern temperature": 29.9,
    "dhw current extern temperature": 56.1,
    "dhw current tap water flow": 0.0,
    "dhw storage intern temperature": 29.9,
    "dhw storage extern temperature": 56.1,
    "dhw activated": "on",
    "dhw one time charging": "off",
    "dhw disinfecting": "off",
    "dhw charging": "off",
    "dhw recharging": "off",
    "dhw temperature ok": "on",
    "dhw active": "off",
    "dhw 3-way valve active": "on",
    "dhw starts": 13210,
    "dhw active time": "0 days 0 hours 0 minutes",
    "nominal Power": 18,
    "total energy": 0.00,
    "energy heating": 0.00,
    "dhw energy": 0.00
  },
  "Thermostat Worcester Comfort RF (DeviceID:0x18, ProductID:215, Version:01.00)": {
    "hc1 selected room temperature": 8.0,
    "hc1 current room temperature": 12.5,
    "hc1 Discovery current room temperature": "roomTemp",
    "hc1 mode": "auto",
    "hc1 mode type": "off",
    "hc1 target flow temperature": 0
  },
  "Controller Module Controller (DeviceID:0x09, ProductID:152, Version:02.09)": {},
  "Custom Entities": {},
  "Analog Sensors": {},
  "Temperature Sensors": {}
}
```

**Reply by jtricerolph** [reply]
Was anything else needed?
I had renamed burngas to oil while tinkering trying to work out what entities I could use to setup something for energy calculations.

**Reply by MichaelDvP** [reply]
Please try one of the builds on https://github.com/MichaelDvP/EMS-ESP32/releases
Seems all burner values are missing in the ems-telegrams, maybe a 3-party burner only started by relais? 
Needs some time to check the logs for missing values and what can be used instead.

**Reply by jtricerolph** [reply]
As I say its oil rather than gas if makes any difference.
The boiler_gasburner goes on when firing for both water and heating but heating active doesn't change
The 3 way valve changes to off while heating running 🤷‍♀️
Is there anything else I can do to help?

**Reply by jtricerolph** [reply]
```
{
  "Boiler Worcester GBx72/Trendline/Cerapur/Greenstar Si (DeviceID:0x08, ProductID:123, Version:06.02)": {
    "force heating off": "off",
    "heating active": "on",
    "tapwater active": "off",
    "selected flow temperature": 70,
    "heating pump modulation": 0,
    "current flow temperature": 56.9,
    "return temperature": 15.4,
    "actual boiler temperature": 59.1,
    "oil": "on",
    "gas stage 2": "off",
    "flame current": 0.0,
    "fan": "off",
    "ignition": "off",
    "oil preheating": "off",
    "burner min power": 0,
    "burner max power": 100,
    "burner min period": 1,
    "hysteresis on temperature": -6,
    "hysteresis off temperature": 4,
    "hysteresis stage 2 on temperature": 0,
    "hysteresis stage 2 off temperature": 0,
    "heating activated": "on",
    "heating temperature": 70,
    "heating pump": "on",
    "boiler pump max power": 100,
    "boiler pump min power": 10,
    "boiler pump mode": "proportional",
    "pump delay": 3,
    "set flow temperature": 90,
    "burner set power": 100,
    "burner selected max power": 0,
    "burner current power": 0,
    "burner starts": 24389,
    "total burner operating time": "0 days 0 hours 0 minutes",
    "burner stage 2 operating time": "0 days 0 hours 0 minutes",
    "total heat operating time": "0 days 0 hours 0 minutes",
    "burner starts heating": 11176,
    "total UBA operating time": "1409 days 5 hours 16 minutes",
    "service code number": 200,
    "maintenance message": "H00",
    "maintenance scheduled": "off",
    "time to next maintenance": 6000,
    "next maintenance date": "01.01.2009",
    "dhw turn on/off": "on",
    "dhw set temperature": 0,
    "dhw selected temperature": 59,
    "dhw solar boiler temperature": 15.4,
    "dhw type": "flow",
    "dhw comfort": "eco",
    "dhw flow temperature offset": 40,
    "dhw max power": 100,
    "dhw circulation pump available": "off",
    "dhw charging type": "3-way valve",
    "dhw hysteresis on temperature": -5,
    "dhw hysteresis off temperature": 0,
    "dhw disinfection temperature": 70,
    "dhw circulation pump mode": "2x3min",
    "dhw circulation active": "off",
    "dhw current intern temperature": 49.5,
    "dhw current extern temperature": 59.2,
    "dhw current tap water flow": 0.0,
    "dhw storage intern temperature": 49.6,
    "dhw storage extern temperature": 59.1,
    "dhw activated": "on",
    "dhw one time charging": "off",
    "dhw disinfecting": "off",
    "dhw charging": "off",
    "dhw recharging": "off",
    "dhw temperature ok": "on",
    "dhw active": "off",
    "dhw 3-way valve active": "off",
    "dhw starts": 13213,
    "dhw active time": "0 days 0 hours 0 minutes",
    "nominal Power": 18,
    "total energy": 0.00,
    "energy heating": 0.00,
    "dhw energy": 0.00
  },
  "Thermostat Worcester Comfort RF (DeviceID:0x18, ProductID:215, Version:01.00)": {
    "date/time": "21.11.2023 01:23",
    "hc1 selected room temperature": 20.0,
    "hc1 current room temperature": 15.6,
    "hc1 Discovery current room temperature": "roomTemp",
    "hc1 mode": "auto",
    "hc1 mode type": "on",
    "hc1 target flow temperature": 90
  },
  "Controller Module Controller (DeviceID:0x09, ProductID:152, Version:02.09)": {},
  "Custom Entities": {},
  "Analog Sensors": {},
  "Temperature Sensors": {}
}
```

**Reply by jtricerolph** [reply]
Heating and tap water active now come on accordingly.

**Reply by MichaelDvP** [reply]
Have to check if this does not break anything for other boilers and add the energy calculation by adding curBurnPower as 100% while burngas(oil) is on. The boiler does not send burner values, only the burngas relais. I've looked in the manual https://www.worcester-bosch.co.uk/professional/support/literature/greenstar-heatslave-ii-internal-installation-instructions, it says that this burner is installed: https://www.rielloburners.co.uk/images/content/downloads/3514157_2903104-16.pdf and it is connected only mains, error-light and rset line. There is no diagnostic., so all burner states (fan, ignition, flame, flamecurrent, currentPower, etc.) in telegram 0x18 are reported zero.

**Reply by jtricerolph** [reply]
Thanks for your help, energy ideally splitting between heating and water /something to monitor on time for a rough fuel consumption were 2 main aims for me 😅
I liked the idea of a non intrusive method rather than other options I've see of tapping in to oil pump circuit to know when oil is pumping 😬

**Reply by jtricerolph** [reply]
@MichaelDvP was there anything else you needed from me? Did you get what you needed from logs?

**Reply by jtricerolph** [reply]
I understand there may be potential conflict with other models so if its not posible to get working could you clarify what values you would have used such that I can replicate and calculate within homeassistant if its not posible to get energy stats working within espems.

**Reply by MichaelDvP** [reply]
Doesn't it work with the latest builds? I thought [this commit](https://github.com/MichaelDvP/EMS-ESP32/commit/b9af4325c9f6165c18699bc49b8f80015654bcfc) fixes it.

**Reply by jtricerolph** [reply]
Apologies, should I update again? The dev update made heating and tap water acrite respond but not energy.
Which version should I use?

**Reply by jtricerolph** [reply]
Development Build v3.6.5-dev.4
That seems to have done it, sorry, didn't realise you have already updated.
Thanks.
If I wanted to setup a rough gague for actual fuel consumption would you convert back from kw for rough figure?

**Reply by MichaelDvP** [reply]
For fuel consumption just divide the energy by 9.8 kWh/l, as mentioned [here](https://github.com/emsesp/EMS-ESP32/discussions/1483#discussioncomment-7764079)
